File: movieApp/app/src/main/java/com/lbc/practice/movieapp/MovieApplication.java
//package com.lbc.practice.movieapp;
//
//import android.graphics.Movie;
//
//import com.lbc.practice.movieapp.data.resource.MovieRepository;
//
//import javax.inject.Inject;
//
//import dagger.android.AndroidInjector;
//import dagger.android.DaggerApplication;
//
//public class MovieApplication extends DaggerApplication {
//
//    @Inject
//    MovieRepository movieRepository;
//
//    @Override
//    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
//        return null;
//    }
//
//    public MovieRepository getMovieRepository (){
//        return movieRepository;
//    }
//}

File: movieApp/app/src/main/java/com/lbc/practice/movieapp/MovieApplication.kt
package com.lbc.practice.movieapp;

import android.graphics.Movie;

import com.lbc.practice.movieapp.data.resource.MovieRepository;
import com.lbc.practice.movieapp.di.DaggerAppComponent

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;


class MovieApplication : DaggerApplication() {

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerAppComponent.builder().application(this).build()
    }

}
